empress_gui.py

```diff
@@ -0,0 +1,335 @@
+        """Plots the cost regions using matplotlib and embeds the graph in a tkinter window."""    
+        # creates a new tkinter window 
+        plt_window = tk.Toplevel(self.master)
+        plt_window.geometry("550x550")
+        plt_window.title("Matplotlib Graph DEMO")
+        # creates a new frame
+        plt_frame = tk.Frame(plt_window)
+        plt_frame.pack(fill=tk.BOTH, expand=1)
+        plt_frame.pack_propagate(False)
+        recon_input = empress.read_input(self.file_path)
+        cost_regions = empress.compute_cost_regions(recon_input, 0.5, 10, 0.5, 10)  
+        #cost_regions.draw_to_file('./examples/cost_poly.png')  # draw and save to a file
+        fig = cost_regions.draw()  # draw to figure (creates matplotlib figure)
+        canvas = FigureCanvasTkAgg(fig, plt_frame)
+        canvas.draw()
+        canvas.get_tk_widget().pack(side=tk.TOP, fill=tk.BOTH, expand=True)
+        # the toolbar allows the user to zoom in/out, drag the graph and save the graph
+        toolbar = NavigationToolbar2Tk(canvas, plt_frame)
+        toolbar.update()
+        canvas.get_tk_widget().pack(side=tk.TOP)
+        # updates the DTL costs using the x,y coordinates clicked by the user inside the graph 
+        # otherwise pops up a warning message window
+        fig.canvas.callbacks.connect('button_press_event', self.get_xy_coordinates)
+
+    def get_xy_coordinates(self, event):
+        """Updates the DTL costs when user clicks on the matplotlib graph, otherwise pops up a warning message window."""
+        if event.inaxes is not None:
+            self.dup_cost.set("1.00")
+            self.trans_cost.set(round(event.ydata, 2))
+            self.loss_cost.set(round(event.xdata, 2))
+            # enables the next step, viewing reconciliation result
+            self.valid_dup_entry = 1.00
+            self.valid_trans_entry = event.ydata
+            self.valid_loss_entry = event.xdata
+            self.enable_recon_btn()
+        else:
+            messagebox.showinfo("Warning", "Please click inside the axes bounds.")
+
+    def check_dup_entry(self, *args):
+        """Checks and updates the duplication cost when user enters a value in the entry box."""
+        try:
+            self.numbers_dup_entry = float(self.dup_cost.get())
+            if self.numbers_dup_entry >= 0:
+                self.valid_dup_entry = self.numbers_dup_entry
+                self.dup_cost.set(round(self.valid_dup_entry, 2))
+                self.enable_recon_btn()
+            else:
+                messagebox.showinfo("Warning", "Please enter a non-negative number.")
+        except:
+            messagebox.showinfo("Warning", "Please enter a non-negative number.")
+    
+    def check_trans_entry(self, *args):
+        """Checks and updates the transfer cost when user enters a value in the entry box."""
+        try:
+            self.numbers_trans_entry = float(self.trans_cost.get())
+            if self.numbers_trans_entry >= 0:
```

Comment:
If you don't care about `numbers_trans_entry` afterwards, don't put it in `self`.

Reply:
I think you probably already know what's the difference between `var` and `self.var`, but if you aren't sure what the difference is, please read some documentation about python classes. https://www.askpython.com/python/python-self-variable https://docs.python.org/3/tutorial/classes.html

Reply:
Ok! Done:)